src/components/pie-chart/index.js

```diff
@@ -80,7 +89,11 @@ const PieChart = ({
     const { finalIndexBy, finalDataKey } = processPieDataKeys({ data, indexBy, dataKey })
     const aggregatedData = aggregateData({ data, keys: [finalDataKey], indexBy: finalIndexBy, groupByKey, valueKey, type: 'sum' })
     const nivoData = convertPieDataToNivo({ data: aggregatedData, indexBy: finalIndexBy, dataKey: finalDataKey })
-    const baseColors = typeof colors === 'function' || !colors.length ? processColors(data.length, colorType, colorParam) : colors
+    let baseColors = colors.length ? colors : processColors(data.length, colorType, colorParam)
+
+    if (typeof colors === 'function') {
+      baseColors = nivoData.map(d => colors(d))
```

Comment:
not overly important but this can be simplified as:

```suggestion
      baseColors = nivoData.map(colors)
```